src/sfdc48/salesforce_query_endpoint.bal

```diff
@@ -0,0 +1,91 @@
+// under the License.
+//
+
+import ballerina/http;
+import ballerina/oauth2;
+
+# Salesforce Client object.
+# + salesforceClient - OAuth2 client endpoint
+# + salesforceConfiguration - Salesforce Connector configuration
+public type QueryClient client object {
+    http:Client salesforceClient;
+    SalesforceConfiguration salesforceConfiguration;
+
+    # Salesforce Connector endpoint initialization function.
+    # + salesforceConfig - Salesforce Connector configuration
+    public function __init(SalesforceConfiguration salesforceConfig) {
+        self.salesforceConfiguration = salesforceConfig;
+        // Create OAuth2 provider.
+        oauth2:OutboundOAuth2Provider oauth2Provider = new(salesforceConfig.clientConfig);
+        // Create bearer auth handler using created provider.
+        http:BearerAuthHandler bearerHandler = new(oauth2Provider);
+
+        http:ClientSecureSocket? socketConfig = salesforceConfig?.secureSocketConfig;
+        
+        // Create salesforce http client.
+        if (socketConfig is http:ClientSecureSocket) {
+            self.salesforceClient = new(salesforceConfig.baseUrl, {
+                secureSocket: socketConfig,
+                auth: {
+                    authHandler: bearerHandler
+                }
+            });
+        } else {
+            self.salesforceClient = new(salesforceConfig.baseUrl, {
+                auth: {
+                    authHandler: bearerHandler
+                }
+            });
+        }
+    }
+
+    //Query
+
+    # Executes the specified SOQL query.
+    # + receivedQuery - Sent SOQL query
+    # + return - `SoqlResult` record if successful else ConnectorError occured
+    public remote function getQueryResult(string receivedQuery) returns @tainted SoqlResult|ConnectorError {
+        string path = prepareQueryUrl([API_BASE_PATH, QUERY], [Q], [receivedQuery]);
+        json res = check self.getRecord(path);
+        return toSoqlResult(res);
+    }
+
+    # If the query results are too large, retrieve the next batch of results using nextRecordUrl.
+    # + nextRecordsUrl - URL to get next query results
+    # + return - `SoqlResult` record if successful else ConnectorError occured
+    public remote function getNextQueryResult(string nextRecordsUrl) returns @tainted SoqlResult|ConnectorError {
+        json res = check self.getRecord(nextRecordsUrl);
+        return toSoqlResult(res);
+    }
+
+    //Search
+
+    # Executes the specified SOSL search.
+    # + searchString - Sent SOSL search query
+    # + return - `SoslResult` record if successful else ConnectorError occured
+    public remote function searchSOSLString(string searchString) returns @tainted SoslResult|ConnectorError {
+        string path = prepareQueryUrl([API_BASE_PATH, SEARCH], [Q], [searchString]);
+        json res = check self.getRecord(path);
+        return toSoslResult(res);
+    }
+
+    public function getRecord(string path) returns @tainted json|ConnectorError {
```

Comment:
Do we need to make this function as public?. and also we need to make this as remote

Reply:
This is more like a util function. Not exposed via the query client. So removing the public privilege should be fine I guess? Or move to a util file.

Reply:
Made this function private.